src/hotspot/share/runtime/javaThread.hpp

```diff
@@ -643,7 +644,12 @@ class JavaThread: public Thread {
   }
 
   bool is_in_VTMS_transition() const             { return _is_in_VTMS_transition; }
+  bool is_in_tmp_VTMS_transition() const         { return _is_in_tmp_VTMS_transition; }
+  bool is_in_any_VTMS_transition() const         { return _is_in_VTMS_transition || _is_in_tmp_VTMS_transition; }
+
   void set_is_in_VTMS_transition(bool val);
+  void toggle_is_in_tmp_VTMS_transition()        { _is_in_tmp_VTMS_transition = !_is_in_tmp_VTMS_transition; };
+
```

Comment:
My suggestion was to have the term "in VTMS transition" be inclusive of temp transitions. So based on your current names I would suggest:

- is_in_VTMS_transition -> is_in_non_tmp_VTMS_transition
- is_in_any_VTMS_transition -> is_in_VTMS_transition

But that becomes a problem for `set_is_in_VTMS_transition`, which would need to be renamed `set_is_in_non_tmp_VTMS_transition`, which I'm guessing you don't want to do. So let's instead just hope this all goes away before thinking about it any more.

Reply:
Thank you for sharing your suggestion.
To be honest, I'm inclined to keep the two as simple as possible, independent end mutually exclusive.
Temporary transitions have big difference comparing to normal transitions.
They are allowed to be suspended and do not clash with VTMS disablers.

Unfortunately, it seems, Alan got some difficulties in getting rid of temporary transitions.
I'll double check on it just to be sure I understand it correctly.

Reply:
ok